baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -0,0 +1,192 @@
+import logging
+import queue
+import socket
+
+from typing import Callable
+from typing import Optional
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+import kombu
+
+from gevent.server import StreamServer
+from kombu.mixins import ConsumerMixin
+from kombu.transport.virtual import Channel
+from pyramid.config import Configurator
+
+from baseplate import Baseplate
+from baseplate import RequestContext
+from baseplate.clients.kombu import KombuSerializer
+from baseplate.frameworks.pyramid import BaseplateConfigurator
+from baseplate.server.queue_consumer import MessageHandler
+from baseplate.server.queue_consumer import PumpWorker
+from baseplate.server.queue_consumer import QueueConsumerFactory
+from baseplate.server.wsgi import make_server as make_wsgi_server
+
+
+if TYPE_CHECKING:
+    WorkQueue = queue.Queue[kombu.Message]  # pylint: disable=unsubscriptable-object
+else:
+    WorkQueue = queue.Queue
+
+
+logger = logging.getLogger(__name__)
+
+
+Handler = Callable[[RequestContext, kombu.Message], None]
+HealthCheck = Callable[[RequestContext], bool]
+
+
+class AbortError(Exception):
```

Comment:
is this just meant for users to use? how about something like `ConsumerFatalError`? worried it's a bit generic of a name.